pkg/statistics/handle/globalstats/topn.go

```diff
@@ -138,7 +262,173 @@ func MergeGlobalStatsTopNByConcurrency(
 //     just delete some values when we merge the global-level topN.
 func MergePartTopN2GlobalTopN(
 	loc *time.Location,
-	version int,
+	topNs []*statistics.TopN,
+	n uint32,
+	hists []*statistics.Histogram,
+	isIndex bool,
+	killer *sqlkiller.SQLKiller,
+) (*statistics.TopN, []statistics.TopNMeta, []*statistics.Histogram, error) {
+	var mergingHeap topnHeap = make([]*heapItem, 0, len(topNs))
+	sumTopN := 0
+	for i, topN := range topNs {
+		if topN.Num() == 0 {
+			continue
+		}
+		sumTopN += topN.Num()
+		heap.Push(&mergingHeap, &heapItem{
+			item:          &topN.TopN[0],
+			idx:           i,
+			nextPosInTopN: 1,
+		})
+	}
+	maxPossibleAdded := make([]int64, len(hists))
+	for i, hist := range hists {
+		curMax := int64(hist.NotNullCount() / float64(hist.NDV))
+		for _, bkt := range hist.Buckets {
+			curMax = max(curMax, bkt.Repeat)
+		}
+		maxPossibleAdded[i] = curMax
+	}
+	if mergingHeap.Len() == 0 {
+		return nil, nil, hists, nil
+	}
+	histIters := make([]histIter, len(hists))
+	for i, hist := range hists {
+		histIters[i].hist = hist
+	}
+	type maintaining struct {
+		affectedTopNs *bitset.BitSet
+		item          statistics.TopNMeta
+	}
+	cur := maintaining{
+		affectedTopNs: bitset.New(uint(len(hists))),
+	}
+	step := int64(0)
+	histRemoveCnt := int64(0)
+	var finalTopNs topNMeataHeap = make([]statistics.TopNMeta, 0, n+1)
+	remainedTopNs := make([]statistics.TopNMeta, 0, n)
+	skipCount := 0
+	affectedHist := make([]int, 0, len(hists))
+	firstTime := true
+	checkTheCurAndMoveForward := func(nextVal *statistics.TopNMeta, position uint) error {
```

Comment:
Maybe better to move this function out. Because it is quite long.

Reply:
If so, I prefer just inlining it, or adding another wrapper or helper.

Reply:
Another wrapper sounds good to me.